### PR TITLE
[FLINK-32938] Replace pulsar admin calls

### PR DESCRIPTION
**Pull request by nlu90**
## Purpose of the change

Replace PulsarAdmin calls with PulsarClient or Consumer calls

## Brief change log

- *Change the internal design of `ProducerRegister`.*
- *Expose topic metadata query in `PulsarSinkContext`.*
- *Change the internal metadata cache in `MetadataListener`.*

## Verifying this change

This change is already covered by existing tests, such as *(please describe tests)*.

## Significant changes

*(Please check any boxes [x] if the answer is "yes". You can first publish the PR and check them afterwards, for
convenience.)*

- [ ] Dependencies have been added or upgraded
- [ ] Public API has been changed (Public API is any class annotated with `@Public(Evolving)`)
- [ ] Serializers have been changed
- [ ] New feature has been introduced
    - If yes, how is this documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by nlu90** [comment]
Can someone help rerun the CI?  On my local laptop, everything runs fine:

```
-> mvn clean verify
...
...
...
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Flink : Connectors : Pulsar : Parent 4.0-SNAPSHOT:
[INFO] 
[INFO] Flink : Connectors : Pulsar : Parent ............... SUCCESS [  1.819 s]
[INFO] Flink : Connectors : Pulsar ........................ SUCCESS [27:41 min]
[INFO] Flink : Connectors : Pulsar : SQL .................. SUCCESS [  5.184 s]
[INFO] Flink : Connectors : Pulsar : E2E Tests ............ SUCCESS [  1.157 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  27:50 min
[INFO] Finished at: 2023-09-11T15:01:51-07:00
[INFO] ------------------------------------------------------------------------

```

**Comment by tisonkun** [comment]
@nlu90 retriggered and passed. I'm reviewing this patch now.

File: flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/sink/PulsarSinkBuilder.java [hunk 1]
```diff
@@ -126,9 +123,11 @@ public class PulsarSinkBuilder<IN> {
      *
      * @param adminUrl The url for the PulsarAdmin.
      * @return this PulsarSinkBuilder.
+     * @deprecated this method will return builder directly
```
Comment (tisonkun) [inline flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/sink/PulsarSinkBuilder.java:126]: Perhaps add a link to this PR as background. But it's OK that we do it in a follow-up patch.

**Review by tisonkun — APPROVED** [review]
LGTM. Thank you!

We need to update the document a bit. And since we bump the minima Pulsar version, I'm afraid that we already need a new major version bump. And thus it's possible we remove the deprecated method.

**Comment by tisonkun** [comment]
Merging...

I'll prepare a follow-up patch for docs updates, which includes a response for FLINK-33111

**Comment by thomasg19930417** [comment]
@nlu90 This pull request has led to the occurrence of the FLINK-35477 issue. When it is impossible to recover from the checkpoint (ck), the consumption starting point is no longer the existing offset but rather from the startCursor.

**Pull request merged**